graphics/tinygl/init.cpp

```diff
@@ -44,7 +44,15 @@ void initSharedState(GLContext *c) {
 void endSharedState(GLContext *c) {
 	GLSharedState *s = &c->shared_state;
 
-	free_texture(c, 0);
```

Comment:
There is a tangential but aggravating problem here: `tglDisposeResources` takes a `GLContext` parameter. As a result, it can only be given a prototype where `zgl.h` can be imported. But `zgl.h` imports `zdirtyrect.h`, and `tglDisposeResources` is defined in `zdirtyrect.cpp` - its prototype hence cannot be defined in `zdirtyrect.h`.

There is the same problem with `zgl.h` vs. `zblit.cpp`, but "lucklily" the implementation only works with the default context - so prototype for `Graphics::Internal::tglCleanupImages` exists.
I could call `Graphics::Internal::tglCleanupImages` from `glClose` - where I think it belongs as the image list is on `GLContext`. By calling `tglDisposeResources` from `endSharedState` it will indeed only free disposed textures (and without duplicating that logic), but then will also call `Graphics::Internal::tglCleanupImages` internally... Which may be overall correct if we came from `glClose` but is inconsistent with the `GLSharedState`/`GLContext` data structure distinction.